traveler-ios-ui/TravelerKitUI/ViewControllers/CancellationViewController.swift

```diff
@@ -15,57 +15,124 @@ protocol CancellationViewControllerDelegate: class {
 }
 
 class CancellationViewController: UITableViewController {
-    @IBOutlet weak var totalRefundLabel: UILabel!
+    @IBOutlet weak var explanationInputTextField: UITextField!
 
     var quote: CancellationQuote?
     weak var delegate: CancellationViewControllerDelegate?
 
+    private var selectedReason: CancellationReason?
+    private var explanation: String?
+    private var selectedReasonIndexPath: IndexPath?
+
     override func viewDidLoad() {
         super.viewDidLoad()
-
-        totalRefundLabel.text = quote?.totalRefund.localizedDescriptionInBaseCurrency
     }
 
     // MARK: UITableViewDataSource
 
     override func numberOfSections(in tableView: UITableView) -> Int {
-        return 1
+        return 2
     }
 
     override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return quote?.products.count ?? 0
+        if section == 0 {
+            return (quote?.products.count ?? 0) + 1
+        } else {
+            return (quote?.cancellationReasons.count ?? 0)
+        }
+    }
+
+    override func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        if section == 1 {
+            return "Please select a reason for cancellation"
+        } else {
+            return nil
+        }
+    }
+
+    override func tableView(_ tableView: UITableView, titleForFooterInSection section: Int) -> String? {
+        if section == 0 {
+            return " "
+        } else {
+            return nil
+        }
     }
 
     override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        let cell = tableView.dequeueReusableCell(withIdentifier: "productCancelCell", for: indexPath) as! InfoCell
+        switch (indexPath.section, indexPath.row == (quote?.products.count ?? 0)) {
```

Comment:
use a more descriptive tuple...use the row itself as oppose to the boolean condition..this is difficult to read